src/main/java/CMV.java

```diff
@@ -118,7 +121,45 @@ private boolean LIC5() {
    * Computes the LIC 6 condition.
    */
   private boolean LIC6() {
-    // TODO: Implementation.
+
+    if (parameters.N_PTS < 3 || parameters.N_PTS > points.length) return false;
+    if (parameters.DIST < 0) return false;
+    if (points.length < 3) return false;
+
+    for (int i = 0; i < points.length - parameters.N_PTS; i++) {
+      Point[] range = Arrays.copyOfRange(points, i, i + parameters.N_PTS);
+      var start = range[0];
+      var end = range[parameters.N_PTS - 1];
+
+      if (start.equals(end)) {
+        // dist = sum of distances to all other points
```

Comment:
Here I'm not sure I've interpreted it the same way you have.

> If the first and last points of these N_PTS are identical, then the calculated distance to compare with DIST will be the distance from the coincident point to all other points of the N_PTS consecutive points.

I think they mean "for each of the other points", i.e:

"distance from the coincident point" to "point1" ?> dist
"distance from the coincident point" to "point2" ?> dist

and so on (instead of summing all the distances). I could be wrong though! I guess as long as we're all on board with what it means it will be fine.

Reply:
I considered that option as well, I honestly don't know how this is supposed to be interpreted. Would it be okay to keep it as it is, or should I change it?

Reply:
I think it's fine as it is, but also fine to change (which I saw you did, sorry I wasn't quick enough to see your message here). I doubt the TA will complain on either interpretation; as you say it's not entirely unambiguous how to interpret it...